c7n/resources/s3.py

```diff
@@ -1924,15 +1924,16 @@ def get_event_buckets(self, client, trails):
         """
         event_buckets = {}
         for t in trails:
-            for events in client.get_event_selectors(
-                    TrailName=t['Name']).get('EventSelectors', ()):
-                if 'DataResources' not in events:
-                    continue
-                for data_events in events['DataResources']:
-                    if data_events['Type'] != 'AWS::S3::Object':
+            if t.get('HomeRegion') == client._client_config.region_name:
```

Comment:
@kapilt - Is this an appropriate line to add?  I was seeing an error where a trail was created in us-east-1 (or at least it was created a bucket that existed in us-east-1), and when I ran the functional test against us-east-2 I would get a traceback:

```
Traceback (most recent call last):
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/tests/test_account.py", line 660, in test_data_events
    resources = p.run()
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/policy.py", line 714, in __call__
    resources = mode.run()
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/policy.py", line 305, in run
    resources = self.policy.resource_manager.resources()
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/query.py", line 397, in resources
    return self.filter_resources(resources)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/manager.py", line 88, in filter_resources
    resources = f.process(resources, event)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/resources/s3.py", line 1932, in process
    event_buckets = self.get_event_buckets(client, trails)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/resources/s3.py", line 1918, in get_event_buckets
    TrailName=t['Name']).get('EventSelectors', ()):
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/lib/python2.7/site-packages/botocore/client.py", line 312, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/lib/python2.7/site-packages/botocore/client.py", line 601, in _make_api_call
    raise error_class(parsed_response, operation_name)
TrailNotFoundException: An error occurred (TrailNotFoundException) when calling the GetEventSelectors operation: Unknown trail: skunk-trails for the user: 644160558196
```

Reply:
not sure that it is (quick diff look), we effectively want to discover data events trail existence regardless of home region on a global trail, the right thing would be to get a new client for the home region of the trail to verify.

Reply:
fixed